src/intentLogic/getOpinion.js

```diff
@@ -3,29 +3,75 @@ const asJson = require('simple-get-promise').asJson;
 
-module.exports = function () {
-    this.event.session.attributes.lastIntent = 'GetOpinionIntent';
+module.exports = function (isNewIntentFlag) {
+    // An intent is a new intent unless this is explicitly set to `false`; `undefined` defaults to `true`.
+    const isNewIntent = isNewIntentFlag !== false;
 
+    const attributes = this.event.session.attributes;
     const slots = this.event.request.intent.slots;
-    const searchTerm = slots.search_term ? slots.search_term.value : null;
 
-    var capi_filter = 'show-fields=standfirst,byline,headline&show-blocks=all&tag=commentisfree/commentisfree';
-    var capi_query = helpers.capiQuery('search', capi_filter, searchTerm);
+    attributes.lastIntent = 'GetOpinionIntent';
+    if (isNewIntent) attributes.searchTerm = slots.search_term ? slots.search_term.value : null;
 
-    get(capi_query)
+    attributes.moreOffset = getMoreOffset(isNewIntent, attributes.moreOffset);
+
+    get(buildCapiQuery(attributes.searchTerm))
         .then(asJson)
-        .then(json => {
-            if (json.response.results && json.response.results.length > 1) {
-                var opinion_speech = randomMsg(speech.acknowledgement) + speech.opinions.latest;
-                opinion_speech += json.response.results[0].fields.headline + ' by ' + json.response.results[0].fields.byline + '. ' + json.response.results[0].blocks.body[0].bodyTextSummary;
-                this.emit(':ask', opinion_speech, speech.opinions.reprompt)
+        .then((json) => {
+            if (json.response.results && json.response.results.length >= attributes.moreOffset) {
+                updatePositionalContent(json); // side effects, yay!
+                this.emit(':ask', generateOpinionSpeech(json));
             } else {
-                this.emit(':tell', speech.opinions.notfound);
+                this.emit(':ask', speech.opinions.notfound);
             }
-        })
-        .catch(error => {
-            this.emit(':tell', speech.opinions.notfound);
-        })
+
+        });
+
+    var generateOpinionSpeech = (json) => {
+        const preamble = generatePreamble(isNewIntent, attributes.searchTerm, json.response.results.length);
+        const conclusion = sound.strongBreak + followupQuestion(json.response.results.length);
+
+        var getOpinions = () => {
+            return json.response.results.slice(attributes.moreOffset, attributes.moreOffset+3).map(results =>
+                results.fields.headline + sound.transition
+            );
+        };
+
+        return preamble + getOpinions() + conclusion;
+    };
+
+    var updatePositionalContent = (json) => {
+        attributes.positionalContent = json.response.results.slice(attributes.moreOffset, attributes.moreOffset+3).map(result =>
+            result.id );
+    };
+};
+
+var buildCapiQuery = (searchTerm) => {
+    return helpers.capiQuery('search', '&show-fields=standfirst,byline,headline&tag=commentisfree/commentisfree', searchTerm);
```

Comment:
True but that logic is almost a copy of what we already for headlines. I would be tempted to keep it the same across multiple intents if possible to reduce the mental overhead of keeping track of how we're doing things how, if that makes sense.